gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-common/src/main/java/io/gravitee/am/gateway/handler/common/vertx/web/handler/impl/internal/mfa/filter/StepUpAuthenticationFilter.java

```diff
@@ -38,22 +39,21 @@ public StepUpAuthenticationFilter(
 
     @Override
     public Boolean get() {
-        // If Adaptive MFA is active and KO (rule == false) we bypass this filter
-        // Because it could return true and skip MFA
-        if (!context.isMfaSkipped() && context.isAmfaActive() && !context.isAmfaRuleTrue()) {
-            return false;
-        }
         String mfaStepUpRule = context.getStepUpRule();
         if (context.isStepUpActive()) {
             context.setStepUpRuleTrue(isStepUpAuthentication(mfaStepUpRule));
         }
-        return !context.isAmfaActive() &&
-                context.isStepUpActive() &&
-                !context.isStepUpRuleTrue() &&
-                (context.isUserStronglyAuth() || context.isMfaSkipped());
+
+        // if strongly auth and StepUp rule is false (SAFE),
+        // then we can bypass MFA Challenge
+        if (context.isUserStronglyAuth()) {
+            return context.isStepUpActive() && !context.isStepUpRuleTrue();
+        }
+
+        return context.isMfaSkipped();
     }
 
     protected boolean isStepUpAuthentication(String selectionRule) {
-        return ruleEngine.evaluate(selectionRule, context.getEvaluableContext(), Boolean.class, false);
+        return ruleEngine.evaluate(selectionRule, context.getEvaluableContext(), Boolean.class, UNSAFE);
```

Comment:
why is it changed?

Reply:
I just use the constant defined in this class. will revert and remove the constant